src/schema/v2/partner/Settings/createPartnerLocationDaySchedulesMutation.ts

```diff
@@ -0,0 +1,129 @@
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+import DayScheduleType from "schema/v2/day_schedule"
+import { ResolverContext } from "types/graphql"
+
+type DayScheduleInput = {
+  day: number
+  startTime?: number
+  endTime?: number
+}
+
+interface Input {
+  partnerId: string
+  locationId: string
+  daySchedules: DayScheduleInput[]
+}
+
+const DayScheduleInputType = new GraphQLInputObjectType({
+  name: "DayScheduleInput",
+  fields: {
+    day: { type: GraphQLInt },
+    startTime: { type: GraphQLInt },
+    endTime: { type: GraphQLInt },
+  },
+})
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "CreatePartnerLocationDaySchedulesSuccess",
+  isTypeOf: (data) => !!data[0]?._id,
+  fields: () => ({
+    daySchedules: {
+      type: new GraphQLList(DayScheduleType),
+      resolve: (result) => result,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "CreatePartnerLocationDaySchedulesFailure",
+  isTypeOf: (data) => data._type === "GravityMutationError",
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "CreatePartnerLocationDaySchedulesOrError",
+  types: [SuccessType, FailureType],
+})
+
+export const CreatePartnerLocationDaySchedulesMutation = mutationWithClientMutationId<
+  Input,
+  any,
+  ResolverContext
+>({
+  name: "CreatePartnerLocationDaySchedules",
+  description: "Creates a new weekly schedule for a partner location",
+  inputFields: {
+    partnerId: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "ID of the partner",
+    },
+    locationId: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "ID of the location",
+    },
+    daySchedules: {
+      type: new GraphQLNonNull(
+        new GraphQLList(new GraphQLNonNull(DayScheduleInputType))
+      ),
+      description: "List of day schedules for the full week",
+    },
+  },
+  outputFields: {
+    daySchedulesOrError: {
+      type: ResponseOrErrorType,
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: async (
+    { partnerId, locationId, daySchedules },
+    { createPartnerLocationDaySchedulesLoader }
+  ) => {
+    if (!createPartnerLocationDaySchedulesLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    try {
+      const identifiers = { partnerId, locationId }
+
+      const formatDayScheduleDataForQueryString = Object.fromEntries(
+        daySchedules.flatMap((schedule, index) => [
+          [`day_schedules[${index}][day]`, schedule.day],
+          [`day_schedules[${index}][start_time]`, schedule.startTime],
+          [`day_schedules[${index}][end_time]`, schedule.endTime],
+        ])
+      )
```

Comment:
Dont feel the best about this ^^

But needed to format the incomming array data into the proper query string format for the loader

Formats like this:
```
{
  "day_schedules[0][day]": 1,
  "day_schedules[0][start_time]": 9,
  "day_schedules[0][end_time]": 17,
  ...
}
```

and then the `fetch.ts` array logic will work [here](https://github.com/artsy/metaphysics/blob/main/src/lib/apis/fetch.ts#L17-L23), def open to feedback here, here is the API endpoint [contract](https://github.com/artsy/gravity/blob/main/app/api/v1/partners_locations_day_schedules_endpoint.rb#L48-L56)

Reply:
i've totally been bit by that query string array format in the past, just pure confusion 😅. Its how it works tho, so 👍